extensions/azurecore/src/azureResource/providers/universal/universalTreeDataProvider.ts

```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as msRest from '@azure/ms-rest-js';
+
+import { AzureResourceErrorMessageUtil } from '../../utils';
+import { GraphData } from '../../interfaces';
+import { ResourceTreeDataProviderBase } from '../resourceTreeDataProviderBase';
+import { AzureAccount, azureResource } from 'azurecore';
+import { Logger } from '../../../utils/Logger';
+import { AzureResourceUniversalService } from './universalService';
+
+export class AzureResourceUniversalTreeDataProvider<S extends GraphData, D extends GraphData>
+	extends ResourceTreeDataProviderBase<S, D> implements azureResource.IAzureUniversalTreeDataProvider {
+
+	public constructor(
+		private universalService: azureResource.IAzureResourceService,
+	) {
+		super(universalService);
+	}
+
+	public override getService(): azureResource.IAzureResourceService {
+		return this.universalService;
+	}
+
+	public override getChildren(element: azureResource.IAzureResourceNode): Promise<azureResource.IAzureResourceNode[]> {
+		throw new Error('Method not supported for universal provider.');
+	}
+
+	public getTreeItemForResource(resource: azureResource.AzureResource, account: AzureAccount): azdata.TreeItem {
+		let service: AzureResourceUniversalService = this.universalService as AzureResourceUniversalService;
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
```

Comment:
It would be great to fix these instead of just disabling the rule

Reply:
I'm trying.. :)